src/or/connection.c

```diff
@@ -3186,6 +3186,34 @@ connection_bucket_adjust(const or_options_t *options)
   }
 }
 
+/**
+ * Cached value of the last coarse-timestamp when we refilled the
+ * global buckets.
+ */
+static uint32_t last_refilled_global_buckets_ts=0;
+/**
+ * Refill the token buckets for a single connection <b>conn</b>, and the
+ * global token buckets as appropriate.  Requires that <b>now_ts</b> is
+ * the time in coarse timestamp units.
+ */
+static void
+connection_bucket_refill_single(connection_t *conn, uint32_t now_ts)
+{
+  /* Note that we only check for equality here: the underlying
+   * token bucket functions can handle moving backwards in time if they
+   * need to. */
+  if (now_ts != last_refilled_global_buckets_ts) {
+    token_bucket_rw_refill(&global_bucket, now_ts);
+    token_bucket_rw_refill(&global_relayed_bucket, now_ts);
+    last_refilled_global_buckets_ts = now_ts;
+  }
+
+  if (connection_speaks_cells(conn) && conn->state == OR_CONN_STATE_OPEN) {
```

Comment:
So hmmm, we don't count bw when negotiating TLS (`state != OPEN`)?

Reply:
(That's right, apparently.)

Reply:
(At least, I don't think we did before, did we?)

Reply:
Hmm, yeah I think we do not looking at `connection_buf_read_from_socket()`